src/core/project/qgsprojectstylesettings.h

```diff
@@ -0,0 +1,145 @@
+/***************************************************************************
+    qgsprojectstylesettings.h
+    ---------------------------
+    begin                : May 2022
+    copyright            : (C) 2022 by Mathieu Pellerin
+    email                : nirvn dot asia at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSPROJECTSTYLESETTINGS_H
+#define QGSPROJECTSTYLESETTINGS_H
+
+#include "qgis_core.h"
+#include "qgstextformat.h"
+#include "qgswkbtypes.h"
+
+#include <memory.h>
+
+class QDomElement;
+class QgsReadWriteContext;
+class QDomDocument;
+class QgsProject;
+class QgsSymbol;
+class QgsColorRamp;
+
+/**
+ * \brief Contains settings and properties relating to how a QgsProject should handle
+ * styling.
+ *
+ * \ingroup core
+ * \since QGIS 3.26
+ */
+class CORE_EXPORT QgsProjectStyleSettings : public QObject
+{
+    Q_OBJECT
+
+  public:
+
+    /**
+     * Constructor for QgsProjectStyleSettings for the specified \a project.
+     *
+     * Ownership is transferred to the \a project.
+     */
+    QgsProjectStyleSettings( QgsProject *project = nullptr );
+
+    /**
+     * Returns the project default symbol for a given geometry type.
+     * \param geomType the geometry type
+     * \returns a symbol pointer or NULL if there is no default set
+     * \note the symbol ownership is transferred to the caller
+     */
+    QgsSymbol *defaultSymbol( QgsWkbTypes::GeometryType geomType ) const;
```

Comment:
I think it would be cleaner if defaultSymbol/setDefaultSymbol used Qgis::SymbolType instead of GeometryType. (Leaving the geometry type -> symbol type conversion logic in QgsSymbol::defaultSymbol alone.)